bpfassets/libbpf/src/kepler.bpf.c

```diff
@@ -180,11 +182,52 @@ static inline u64 get_on_cpu_cache_miss(u32 *cpu_id)
 	return delta;
 }
 
+static inline void register_new_process_if_not_exist()
+{
+	u64 cgroup_id;
+	u64 pid_tgid = bpf_get_current_pid_tgid();
+	u32 curr_pid = (u32)pid_tgid;
+	u32 curr_tgid = pid_tgid >> 32;
+
+	// create new process metrics
+	struct process_metrics_t *curr_tgid_metrics =
+		bpf_map_lookup_elem(&processes, &curr_tgid);
+	if (!curr_tgid_metrics) {
+		cgroup_id = bpf_get_current_cgroup_id();
+		// the Kernel tgid is the user-space PID, and the Kernel pid is the
+		// user-scape TID
+		process_metrics_t new_process = {
+			.pid = curr_tgid,
```

Comment:
Thread Group ID (TGID) in Kernel is the PID in user space.
On the other hand, the PID in Kernel space is the the thread ID in user space.
See: 
- https://github.com/iovisor/bcc/pull/674
- https://github.com/iovisor/bcc/blob/master/tools/offcputime.py#L69

So that, we are interested in in the PID in user space, which is the TGID in the Kernel.

Reply:
In fact, for simplicity's sake, I don't think we need to export the Kernel space PID (user TID). 
We can just have the user PID (Kernel TIGID) in the golang code.